packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -4313,6 +4313,48 @@ void main() {
     expect(formatter.log, referenceLog);
   });
 
+  testWidgets('formatter logic handles initial repeat edge case', (WidgetTester tester) async {
```

Comment:
Maybe add a comment saying this is a regression test for https://github.com/flutter/flutter/issues/53612?